chalk-recursive/src/recursive.rs

```diff
@@ -163,18 +163,6 @@ impl<'me, I: Interner> Solver<'me, I> {
                 return *minimums;
             }
 
-            let old_answer = &self.context.search_graph[dfn].solution;
```

Comment:
It might have been a mistake to remove this part. My primary motivation to do so was that your comment [here ](https://github.com/rust-lang/chalk/issues/399#issuecomment-643420016) described it as necessary. In addition, without removing it most tests simply failed. Despite this, I share your concerns about infinite recursion and will try to look into how to integrate it with the other changes (i.e. I suspect that the change of assumed start values would need some adjustment to the combination logic).

Reply:
There's a discussion about this [here](https://github.com/rust-lang/chalk/pull/526#discussion_r440336092), but I already couldn't remember the specific case that made this necessary back then :grimacing:

Reply:
I think in principle it should be unnecessary -- but I think you had a legit concern about something that made it sort of non-obvious that it was unnecessary. Grr.

Reply:
FWIW, I compiled rust-analyzer with this PR, ran it on a few examples and didn't observe any difference in results or performance, so that points to it indeed being unnecessary.